git-config/src/parse.rs

```diff
@@ -0,0 +1,139 @@
+use crate::{file, spanned};
+use dangerous::{Bytes, BytesReader, Error, Expected, Input};
+
+fn config(bytes: &[u8]) -> Result<Vec<file::Token>, Expected<'_>> {
+    fn config<'i, E>(input: Bytes<'i>, r: &mut BytesReader<'i, E>) -> Result<Vec<file::Token>, E>
+    where
+        E: Error<'i>,
+    {
+        let mut tokens = Vec::new();
+        skip_whitespace_or_comment(r, ConsumeTo::NextToken).map(|section| {
+            tokens.push(spanned::Comment(
+                dangerous::input(section)
+                    .span_of(&input)
+                    .expect("range contained")
+                    .into(),
+            ))
+        });
+        unimplemented!("sections and values");
+    }
+    let input = dangerous::input(bytes);
+    input.read_all(|r| config(dangerous::input(bytes), r))
+}
+
+enum ConsumeTo {
+    NextToken,
+    EndOfLine,
+}
+
+#[must_use]
+fn skip_whitespace_or_comment<'a, E>(r: &mut BytesReader<'a, E>, to_where: ConsumeTo) -> Option<&'a [u8]> {
```

Comment:
Yes, this was a little lazy, and this should really be refactored to not need the inner function. One would probably end up with something cloning the input, but… I don't actually know how to do this properly.